SquAPI.lua

```diff
@@ -73,1100 +68,1238 @@ squapi.autoFunctionUpdates = true
-		tailSegmentList = {tailSegmentList}
-	end
-	assert(type(tailSegmentList) == "table", 
-	"your tailSegmentList table seems to to be incorrect")
-	
-    self.berps = {}
-    self.targets = {}
-    self.stiffness = stiffness or .005
-    self.bounce = bounce or .9
-    self.downLimit = downLimit or -90
-    self.upLimit = upLimit or 45
-	for i = 1, #tailSegmentList do
-		assert(tailSegmentList[i]:getType() == "GROUP",
-		"§4The tail segment at position "..i.." of the table is not a group. The tail segments need to be groups that are nested inside the previous segment.§c")
-        self.berps[i] = {squassets.BERP:new(self.stiffness, self.bounce), squassets.BERP:new(self.stiffness, self.bounce, self.downLimit, self.upLimit)}
-        self.targets[i] = {0, 0}
-    end
 
-    self.tailSegmentList = tailSegmentList
-    self.idleXMovement = idleXMovement or 15
-    self.idleYMovement = idleYMovement or 5
-    self.idleXSpeed = idleXSpeed or 1.2
-    self.idleYSpeed = idleYSpeed or 2
-    self.bendStrength = bendStrength or 2
-    self.velocityPush = velocityPush or 0
-    self.initialMovementOffset = initialMovementOffset or 0
-    self.flyingOffset = flyingOffset or 90
-    self.offsetBetweenSegments = offsetBetweenSegments or 1
-    
-
-    -- CONTROL -------------------------------------------------------------------------
-
-    -- UPDATES -------------------------------------------------------------------------
-	
-	self.currentBodyRot = 0
-	self.oldBodyRot = 0
-	self.bodyRotSpeed = 0
-	
-    function self:tick()
-		self.oldBodyRot = self.currentBodyRot
-		self.currentBodyRot = player:getBodyYaw()
-		self.bodyRotSpeed = math.max(math.min(self.currentBodyRot-self.oldBodyRot, 20), -20)
-
-        local time = world.getTime()
-		local vel = squassets.forwardVel()
-		local yvel = squassets.verticalVel()
-		local svel = squassets.sideVel()
-		local bendStrength = self.bendStrength/(math.abs((yvel*30))+vel*30 + 1)
-        local pose = player:getPose()
-	
-        for i = 1, #self.tailSegmentList do
-            self.targets[i][1] = math.sin((time * self.idleXSpeed)/10 - (i)) * self.idleXMovement
-            self.targets[i][2] = math.sin((time * self.idleYSpeed)/10 - (i * self.offsetBetweenSegments) + self.initialMovementOffset) * self.idleYMovement
-
-            self.targets[i][1] = self.targets[i][1] + self.bodyRotSpeed*self.bendStrength + svel*self.bendStrength*40
-			self.targets[i][2] = self.targets[i][2] + yvel * 15 * self.bendStrength - vel*self.bendStrength*15*self.velocityPush
-
-			if i == 1 then
-				if pose == "FALL_FLYING" or pose == "SWIMMING" or player:riptideSpinning() then
-					self.targets[i][2] = self.flyingOffset
-				end	
-			end
-			
-        end
+---TAIL PHYSICS - This will add physics to your tails when you spin, move, jump, etc. Has the option to have an idle tail movement, and can work with a tail with any number of segments.
+---@param tailSegmentList table<ModelPart> The list of each individual tail segment of your tail.
```

Comment:
`table<K, V>` is a type that takes in two parameters. If you meant for this to be an array of parts, use `ModelPart[]` instead.
This parameter also accepts a single model part (which is automatically converted), though it is optional if you wish to show this or not.
```suggestion
---@param tailSegmentList ModelPart[] The list of each individual tail segment of your tail.
```